### PR TITLE
feat(annotation): handle create event

### PR DESCRIPTION
**Pull request by ConradJChan**
Updated the create event to `annotations_create` and added logic to `toggleAnnotationMode` and set the active annotation id in the annotator

Todo:
 - [x] unit tests

### CODE REVIEW

File: src/lib/AnnotationControls.ts [hunk 1]
```diff
@@ -94,6 +94,14 @@ export default class AnnotationControls {
      */
     private handleFullscreenExit = (): void => this.handleFullscreenChange(false);
 
+    /**
+     * Returns the current active annotation mode
+     * @return {string} Current active annotation mode
```
Comment (mxiao6) [inline src/lib/AnnotationControls.ts:99]: The type is `AnnotationMode` right? And I don't think js doc is needed if we have `AnnotationMode` as return type below.

File: src/lib/viewers/BaseViewer.js [hunk 2]
```diff
@@ -1100,6 +1101,18 @@ class BaseViewer extends EventEmitter {
         this.emit('annotatorevent', data);
     }
 
+    handleAnnotationCreateEvent({ annotation, meta: { status } = {} }) {
+        if (status !== 'pending') {
+            const activeMode = this.annotationControls.getActiveMode();
+            this.annotator.toggleAnnotationMode(activeMode);
```
Comment (mxiao6) [inline src/lib/viewers/BaseViewer.js:1107]: Should we do this inside box-annotations?

File: src/lib/viewers/BaseViewer.js [hunk 3]
```diff
@@ -1100,6 +1101,18 @@ class BaseViewer extends EventEmitter {
...
+            this.annotationControls.resetControls();
+        }
+
+        if (status === 'success') {
+            this.annotator.emit('annotations_active_set', annotation.id);
```
Comment (mxiao6) [inline src/lib/viewers/BaseViewer.js:1112]: You can get `id` directly in the function signature;

Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:1112]: Is it possible that the user may click on another annotation during the save XHR? Do we want to try to respect that or is it an edge case we're not worried about right now?

File: src/lib/viewers/BaseViewer.js [hunk 4]
```diff
@@ -1100,6 +1101,18 @@ class BaseViewer extends EventEmitter {
         this.emit('annotatorevent', data);
     }
 
+    handleAnnotationCreateEvent({ annotation, meta: { status } = {} }) {
+        if (status !== 'pending') {
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:1105]: What would the experience be on a failure? Would the unsaved annotation be lost?

File: src/lib/AnnotationControls.ts (hunk 1 above)
Comment (ConradJChan) [reply src/lib/AnnotationControls.ts:99]: Sounds good, I'll remove it then

File: src/lib/viewers/BaseViewer.js (hunk 4 above)
Comment (ConradJChan) [reply src/lib/viewers/BaseViewer.js:1105]: That's a good question, according to the designs, it looks like the failed annotation remains -- I'll follow up with design

File: src/lib/viewers/BaseViewer.js (hunk 2 above)
Comment (ConradJChan) [reply src/lib/viewers/BaseViewer.js:1107]: Given how we're trying to balance box-annotations being a full blown app vs sdk, I think I would lean towards this logic to be determined by the users of the SDK

File: src/lib/viewers/BaseViewer.js (hunk 3 above)
Comment (ConradJChan) [reply src/lib/viewers/BaseViewer.js:1112]: I guess it's possible? I think for now we can leave it, I'll follow up with design, my guess is we may want to disable interactivity during the `isPending` state?

Comment (jstoffan) [reply src/lib/viewers/BaseViewer.js:1112]: Probably just leave it as-is for now after validating with design.

**Pull request merged**